### ISSUE TITLE
Operator ends up in crash loop if secret was removed

### ISSUE THREAD

**Issue by wtrocki** [redhat-developer/service-binding-operator#1024]
### What is the SBO version used?

0.10.0

### What are the steps to reproduce this issue?
  1. Create CR
  2. Remove secret
  3. Create binding
  4. Status will look as follows (but operator will crash)
 
```
conditions:
    - lastTransitionTime: '2021-09-17T13:05:51Z'
      message: secrets "<invalid Value>" not found
      reason: ErrorReadingBinding
      status: 'False'
      type: CollectionReady
    - lastTransitionTime: '2021-09-17T13:05:51Z'
      message: secrets "<invalid Value>" not found
      reason: ProcessingError
      status: 'False'
      type: Ready
  secret: ''
```

### What is the expected behaviour?

Operator will stay working 

### What is the actual behaviour?

Operator ends up in crash loop.

### Service Binding Operator Logs

```
goroutine 420 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x16166c0, 0xc000c454a0)
	/usr/lib/golang/src/runtime/panic.go:969 +0x1b9
reflect.valueInterface(0x0, 0x0, 0x0, 0xc000c0e001, 0x10, 0x10)
	/usr/lib/golang/src/reflect/value.go:1021 +0x1ad
reflect.Value.Interface(...)
	/usr/lib/golang/src/reflect/value.go:1016
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.collectItems(0x0, 0x0, 0x1a10940, 0xc0004c1e00, 0x19faa60, 0xc000132180, 0x15cde00, 0xc000846720, 0x98, 0x0, ...)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:272 +0x44c
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.BindingItems(0x1a10940, 0xc0004c1e00)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:93 +0x376
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline.HandlerFunc.Handle(0x18b2378, 0x1a10940, 0xc0004c1e00)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/api.go:168 +0x3a
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder.(*impl).Process(0xc00077f5c0, 0x17d1380, 0xc0007266c0, 0x0, 0x0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder/pipeline.go:25 +0xd8
github.com/redhat-developer/service-binding-operator/controllers.(*BindingReconciler).Reconcile(0xc0004a8640, 0xc0004fdf99, 0x7, 0xc000556480, 0x1c, 0xc00072a120, 0xc000551e60, 0xc00080cd18, 0xc00080ccf0)
	/tmp/github.com/redhat-developer/service-binding-operator/controllers/common.go:103 +0x622
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00029bef0, 0x1691c40, 0xc00075b140, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go
```

**Comment by sadlerap** [comment]
I've been trying to reproduce this crash, but I haven't been successful yet.  Is there a more concrete example I can use?

I suspect there's an issue with `BindingItems`, but until I can reproduce this, I can't be certain.

**Comment by pedjak** [comment]
It should be fixed by #1032, please reopen if happens again.
